### PR TITLE
[GCI 2019] Added Layout Management to Toner

### PR DESCRIPTION
**Pull request by trungnt2910**
I've added Layout Management to Toner.
Toner now supports larger (and smaller) system font size.
To achieve this, I need to get rid of the preloaded Bitmap, which contains hard-printed text, which cannot be resized. Also, the Bitmap will not look nice if the window resizes.
Also, I also need to add Layout Management directly to the window, so that the window can automatically resize.
This makes the TonerView a bit useless, so I've merged the two files together, just like some other apps, which only has a Window.cpp file.
If there are any functionality problems that may be caused by the merging of the files, let me know, so that I can close this PR and make another request from my unchanged master branch.

### CODE REVIEW

**Comment by trungnt2910** [comment]
@humdingerb I've created the PR here.

File: Makefile [hunk 1]
```diff
@@ -28,20 +28,29 @@ APP_MIME_SIG = application/x-vnd.Be-BTL.Toner
...
+	 TonePlayer.cpp  \
+	 TonerApp.cpp  \
+	 TonerWindow.cpp  \
+
 
 #	Specify the resource definition files to use. Full or relative paths can be
 #	used.
-RDEFS = Toner.rdef
+RDEFS = \
+	 Toner.rdef  \
```
Comment (humdingerb) [inline Makefile]: No reasonn to prepare for more lines here, so I suggest moving the rdef directly into the RDEFS line ybove

File: Makefile [hunk 2]
```diff
@@ -28,20 +28,29 @@ APP_MIME_SIG = application/x-vnd.Be-BTL.Toner
...
+
 
 #	Specify the resource files to use. Full or relative paths can be used.
 #	Both RDEFS and RSRCS can be utilized in the same Makefile.
-RSRCS = 
+RSRCS = \
```
Comment (humdingerb) [inline Makefile]: Remove the "\" and empty line below.

File: Makefile [hunk 3]
```diff
@@ -28,20 +28,29 @@ APP_MIME_SIG = application/x-vnd.Be-BTL.Toner
...
+
 
 # End Pe/Eddie support.
 # @<-src@
 #%}
 
+#%}
```
Comment (humdingerb) [inline Makefile]: Some paste error?

File: TonerWindow.cpp [hunk 4]
```diff
@@ -10,14 +10,43 @@
...
-TonerWindow::TonerWindow() : BWindow(BRect(100, 100, 220, 170), "Toner v1.0", B_TITLED_WINDOW, B_NOT_RESIZABLE | B_WILL_ACCEPT_FIRST_CLICK)
+enum 
 {
-	TonerView *theView = new TonerView( Bounds() );
-	AddChild(theView);
+	Cmd_Level	= 'cm00',
+	Cmd_Freq,
+};
+
+TonerWindow::TonerWindow() : BWindow(BRect(100, 100, 250, 180), "Toner v1.0", B_TITLED_WINDOW, B_NOT_RESIZABLE | B_WILL_ACCEPT_FIRST_CLICK |  B_AUTO_UPDATE_SIZE_LIMITS)
```
Comment (humdingerb) [inline TonerWindow.cpp]: While Tone does not adhere to Haiku's coding style, these very long lines are abit awkward. How about adding a linebreak?
Also, let's remove the "v1.0" from the window title.

File: TonerWindow.cpp [hunk 5]
```diff
@@ -26,3 +55,90 @@ TonerWindow::QuitRequested(void)
...
+	}
+	
+ 	mnuLevel = new BMenuField("Level:", menu);
+	
+	/*    ----------------    The Tone Menu    --------------           */
+
+	menu = new BPopUpMenu("400 Hz");
+	
+	int Tone_Count = 14;
+	const char *ToneText[] = {"20 Hz", "30 Hz", "40 Hz", "60 Hz", "80 Hz", "120 Hz", "240 Hz", "480 Hz", "1 kHz", "2 kHz", "4 kHz", "8 kHz", "16 kHz", "Pink"};
```
Comment (humdingerb) [inline TonerWindow.cpp]: Again a very long line that would benefit from a linebreak.

File: TonerWindow.cpp [hunk 6]
```diff
@@ -10,14 +10,43 @@
...
+};
+
+TonerWindow::TonerWindow() : BWindow(BRect(100, 100, 250, 180), "Toner v1.0", B_TITLED_WINDOW, B_NOT_RESIZABLE | B_WILL_ACCEPT_FIRST_CLICK |  B_AUTO_UPDATE_SIZE_LIMITS)
+{
+	Player = new TonePlayer();
+	Player->SetNoise(false);
+	Player->SetFrequency(400);
+	Player->SetLevel(LEVEL_MUTE);
+		
+	//Create our menus
```
Comment (humdingerb) [inline TonerWindow.cpp]: Since the function "CreateMenus()" is so aptly named, I think we can do without this comment and the surrounding empty  lines.

**Review by humdingerb — CHANGES_REQUESTED** [review]
There are quite a few stray tabs around. You can easily spot them with Koder (set to "Highlight trailing whitespaces").

File: Makefile (hunk 1 above)
Comment (trungnt2910) [reply Makefile]: I've never touched that part of the Makefile, maybe that's generated before.

File: Makefile (hunk 3 above)
Comment (trungnt2910) [reply Makefile]: I've never pasted anything there. Also, it's a comment, so it doesn't affect the build process.
Removed that anyway.

**Comment by trungnt2910** [comment]
I've applied and tested the fixed files, and confirmed that the source still builds properly.

**Comment by trungnt2910** [comment]
Thanks for mentioning Koder, it seems much better than PE

**Review by humdingerb — APPROVED** [review]

**Comment by humdingerb** [comment]
Looks good to me. I'll leave the PR open a bit longer, in case another mentor has more suggestions.

**Comment by trungnt2910** [comment]
Thanks.

**Pull request merged**